mmgen/models/architectures/biggan/generator_discriminator.py

```diff
@@ -0,0 +1,532 @@
+        elif callable(noise):
+            noise_generator = noise
+            assert num_batches > 0
+            noise_batch = noise_generator((num_batches, self.noise_size))
+        # otherwise, we will adopt default noise sampler.
+        else:
+            assert num_batches > 0
+            noise_batch = torch.randn((num_batches, self.noise_size))
+
+        if self.num_classes == 0:
+            label_batch = None
+
+        elif isinstance(label, torch.Tensor):
+            assert label.ndim == 1, ('The label shoube be in shape of (n, )'
+                                     f'but got {label.shape}.')
+            label_batch = label
+        elif callable(label):
+            label_generator = label
+            assert num_batches > 0
+            label_batch = label_generator((num_batches))
+        else:
+            assert num_batches > 0
+            label_batch = torch.randint(0, self.num_classes, (num_batches, ))
+
+        # dirty code for putting data on the right device
+        noise_batch = noise_batch.to(get_module_device(self))
+        if label_batch is not None:
+            label_batch = label_batch.to(get_module_device(self))
+            class_vector = self.shared_embedding(label_batch)
+        else:
+            class_vector = None
+        # If 'split noise', concat class vector and noise chunk
+        if self.split_noise:
+            zs = torch.split(noise_batch, self.noise_chunk_size, 1)
+            z = zs[0]
+            if class_vector is not None:
+                ys = [torch.cat([class_vector, item], 1) for item in zs[1:]]
+            else:
+                ys = zs[1:]
+        else:
+            ys = [class_vector] * len(self.conv_blocks)
+            z = noise_batch
+
+        # First linear layer
+        x = self.noise2feat(z)
+        # Reshape
+        x = x.view(x.size(0), -1, self.input_scale, self.input_scale)
+
+        # Loop over blocks
+        counter = 0
+        for index, conv_block in enumerate(self.conv_blocks):
+            # Second inner loop in case block has multiple layers
+            if isinstance(conv_block, SelfAttentionBlock):
+                x = conv_block(x)
+            else:
+                x = conv_block(x, ys[counter])
+                counter += 1
+
+        # Apply batchnorm-relu-conv-tanh at output
+        out_img = torch.tanh(self.output_layer(x))
+
+        if return_noise:
+            output = dict(
+                fake_img=out_img, noise_batch=noise_batch, label=label)
+            return output
+
+        return out_img
+
+    def init_weights(self, pretrained=None, init_type='ortho'):
+        """Init weights for models.
+
+        We just use the initialization method proposed in the original paper.
+
+        Args:
+            pretrained (str, optional): Path for pretrained weights. If given
+                None, pretrained weights will not be loaded. Defaults to None.
+        """
+        if isinstance(pretrained, str):
```

Comment:
We may support load pre-trained model like this:
https://github.com/open-mmlab/mmgeneration/blob/master/mmgen/models/architectures/stylegan/generator_discriminator_v2.py#L211

Because there are many works using pretrained BigGAN generator and BigGAN discriminator.